tensorflow/compiler/mlir/xla/transforms/hlo_legalize_to_lhlo.cc

```diff
@@ -409,37 +381,31 @@ class StdToLhloReturnOpConverter : public OpConversionPattern<mlir::ReturnOp> {
     auto funcOp = returnOp.getParentOfType<FuncOp>();
     auto numFuncArgs = funcOp.getNumArguments();
     auto loc = returnOp.getLoc();
-
+    auto block = returnOp.getOperation()->getBlock();
     for (auto operand : llvm::enumerate(operands)) {
       auto returnArgNumber = numFuncArgs - numReturnValues + operand.index();
       auto dstBuffer = funcOp.getArgument(returnArgNumber);
       if (dstBuffer == operand.value()) {
         continue;
       }
 
-      auto dealloc = FindInsertionPointForCopy(operand.value());
-
-      if (dealloc == nullptr) {
-        returnOp.emitOpError()
-            << "Missing dealloc for operand " << operand.index();
-        return matchFailure();
-      }
-      OpBuilder::InsertionGuard guard(rewriter);
-      rewriter.setInsertionPoint(dealloc);
+      rewriter.setInsertionPointToEnd(block);
       rewriter.create<xla_lhlo::CopyOp>(loc, llvm::None, operand.value(),
                                         funcOp.getArgument(returnArgNumber));
     }
-    rewriter.replaceOpWithNewOp<xla_lhlo::TerminatorOp>(returnOp);
+    rewriter.setInsertionPointToEnd(block);
```

Comment:
Why can this no longer use `replaceOpWithNewOp`?

Reply:
The generic version of this conversion pattern moved to [BufferAssignment](https://github.com/tensorflow/tensorflow/pull/37212). Your suggestion has also been addressed.